src/main/java/io/confluent/connect/jdbc/dialect/Db2DatabaseDialect.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.connect.jdbc.dialect;
+
+import org.apache.kafka.common.config.AbstractConfig;
+import org.apache.kafka.connect.data.Date;
+import org.apache.kafka.connect.data.Decimal;
+import org.apache.kafka.connect.data.Time;
+import org.apache.kafka.connect.data.Timestamp;
+
+import java.util.Collection;
+
+import io.confluent.connect.jdbc.dialect.DatabaseDialectProvider.SubprotocolBasedProvider;
+import io.confluent.connect.jdbc.sink.metadata.SinkRecordField;
+import io.confluent.connect.jdbc.util.ColumnId;
+import io.confluent.connect.jdbc.util.ExpressionBuilder;
+import io.confluent.connect.jdbc.util.ExpressionBuilder.Transform;
+import io.confluent.connect.jdbc.util.IdentifierRules;
+import io.confluent.connect.jdbc.util.TableId;
+
+/**
+ * A {@link DatabaseDialect} for IBM DB2.
+ */
+public class Db2DatabaseDialect extends GenericDatabaseDialect {
+
+  /**
+   * The provider for {@link Db2DatabaseDialect}.
+   */
+  public static class Provider extends SubprotocolBasedProvider {
+    public Provider() {
+      super(Db2DatabaseDialect.class.getSimpleName(), "db2", "db2j", "ibmdb");
+    }
+
+    @Override
+    public DatabaseDialect create(AbstractConfig config) {
+      return new Db2DatabaseDialect(config);
+    }
+  }
+
+  /**
+   * Create a new dialect instance with the given connector configuration.
+   *
+   * @param config the connector configuration; may not be null
+   */
+  public Db2DatabaseDialect(AbstractConfig config) {
+    super(config, new IdentifierRules(".", "\"", "\""));
```

Comment:
The `GenericDatabaseDialect` serves as a base class that implements everything necessary for the JDBC source connector and most methods necessary for the JDBC sink connector. Thus, dialect implementations only implement the methods that are necessary to customize the behavior of the `GenericDatabaseDialect` class.

One of the things the `GenericDatabaseDialect` class does is use the supplied "identifier rules" that dictate the segment delimiter string, leading quote string, and trailing quote string used in the DBMS' fully-qualified identifiers. DB2 uses the period (`.`) for delimiters, and double quotes for leading and trailing quotes. With these rules, the `GenericDatabaseDialect` can build fully-qualified table names used in SQL statements as well as parse strings into fully-qualified names, allowing the connector user to now support either simple table names and fully-qualified names in the whitelist and blacklist filters; if needed, the names can optionally be quoted using the DBMS-specific quoting characters.